rest-api/src/main/java/gov/cms/qpp/conversion/api/services/CpcFileServiceImpl.java

```diff
@@ -29,6 +32,18 @@
 		return transformMetaDataToUnprocessedCpcFileData(metadata);
 	}
 
+	/**
+	 * Retrieves the file location id and uses it to retrieve the file
+	 *
+	 * @param fileId {@link Metadata} identifier
+	 * @return file returned as an {@link InputStream}
+	 */
+	public InputStream getFileById(String fileId) {
+		String fileLocationId = dbService.getFileSubmissionLocationId(fileId);
```

Comment:
There's no check as to whether this file is a CPC+ file.  The user of this endpoint could guess a UUID, supply it, and get a non-CPC+ file back.  We should somehow verify that only CPC+ files are retrieved through this endpoint.  If what is asked for is not a CPC+ file, throw some exception and return to the caller an appropriate `4xx` status code.

Reply:
Wouldn't this require a scan of the metadata table? If so maybe we should consider tagging cpc+ files.

Reply:
It's fine. I'm glad you caught it because I completely overlooked this here but not for Endpoint 3